mailpoet/lib/WooCommerce/Subscription.php

```diff
@@ -223,6 +223,7 @@ public function handleSubscriberOptin(SubscriberEntity $subscriber, bool $checko
     $signupConfirmation = $this->settings->get('signup_confirmation');
 
     if (!$checkoutOptin) {
+      $this->wp->doAction('mailpoet_woocommerce_segment_unsubscribed', $subscriber);
```

Comment:
As you mentioned in the PR description, when https://mailpoet.atlassian.net/browse/MAILPOET-4178 is implemented, subscribers won't be subscribed anymore during the checkout if the checkbox is unchecked. I'm not sure if we need to implement support for this use case since it is going to be removed soon. Curious to hear what you think. I'm ok to leave it in place if you prefer.

Reply:
Thanks for mentioning it, I was not sure when it would be implemented.
If the other issue is going to be ready soon, I can remove this commit.
@NeosinneR thoughts?

Reply:
It is not a trivial ticket but unless we hit a blocker, it should be ready next week I guess. But happy to proceed with this PR as is. Let's wait for Ján's input.

Reply:
@rodrigoprimo if you feel your ticket will be released in the next week or two, @brezocordero can remove that case from this PR, which would only leave "Customer globally unsubscribes to any marketing emails" - that's the major one anyway.

Even in the optimistic scenario, MailPoet won't be bundled in Woo Express this month, so it's safe to assume that by the time MailPoet will, you won't be able to unsubscribe on the checkout - no point in adding new code around that.

Reply:
I think it will, but at this point, I'm not 100% sure.

I am unsure regarding the best approach. If it is easier, we can merge this PR as is, and I can remove the code related to handling subscriptions during checkout.

Alternatively, we can mark the ticket related to this PR as blocked by https://mailpoet.atlassian.net/browse/MAILPOET-4178. If I manage to finish MAILPOET-4178 in the next week or two, we merge this PR without the code to handle subscriptions during checkout.

Happy to proceed however you prefer, @brezocordero.

Reply:
@rodrigoprimo 
I think it might be easier to merge it and then remove the action hook when you work on your ticket.
You might find that there is a case when we do have to unsubscribe a customer to the WooCommerce segment and in that case we already have the structure for the integration.

Reply:
There is also an acceptance test for when they opt-in/opt-out with AutomateWoo.
It will be good to keep the test and change the outcome to reflect the changes on your ticket.

Reply:
Sounds good to me. Moving this PR to QA.